tests/hardware/stepper_motor/test_stepper_motor_base.py

```diff
@@ -0,0 +1,126 @@
+"""Tests for the StepperMotorBase class."""
+from typing import Any, Optional, Sequence
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from finesse.config import STEPPER_MOTOR_TOPIC
+from finesse.hardware.stepper_motor.stepper_motor_base import StepperMotorBase
+
+
+class _MockStepperMotor(StepperMotorBase):
+    def __init__(self) -> None:
+        self._steps_per_rotation = 1
+        self._step = 0
+        super().__init__()
+
+    @property
+    def steps_per_rotation(self) -> int:
+        return self._steps_per_rotation
+
+    @property
+    def step(self) -> int:
+        return self._step
+
+    @step.setter
+    def step(self, step: int) -> None:
+        self._step = step
+
+    def close(self) -> None:
+        pass
+
+    def stop_moving(self) -> None:
+        pass
+
+    def wait_until_stopped(self, timeout: Optional[float] = None) -> None:
+        pass
+
+    def notify_on_stopped(self) -> None:
+        pass
+
+
+@pytest.fixture
+def stepper() -> _MockStepperMotor:
+    """Provides a basic StepperMotorBase."""
+    return _MockStepperMotor()
```

Comment:
I've never tried, but can you use the `fixture` decorator directly with the class?

Reply:
That settles the question, then!